### ISSUE TITLE
Using Depth/Normal to help a network disentangle lighting.

### ISSUE THREAD

**Issue by pureexe** [pureexe/sd-light-time#7]
Idea: Apply controlnet on training

**Comment by pureexe** [comment]
We apply depth-controlnet on UNet during training, and the image seems to collapse while training.

![image](https://github.com/user-attachments/assets/4237c552-d583-4487-b708-421e82d885bf)

However, when we apply gate_multiply to learning rate (10 left, 100 right) 
It is still not collapsed yet.

![image](https://github.com/user-attachments/assets/9757b05a-2dde-4f0e-87b8-beca7bd7e14c)

**Comment by pureexe** [comment]
update: with Normal & Depth condition, 1e-4 and gate_multiplyer 10 is not working. So, it not the problem of gate multiplier ?

![image](https://github.com/user-attachments/assets/efe793fa-50bb-49d4-914f-29e8ec44611e)

**Comment by pureexe** [comment]
Update:

For conditioning with normal

1e-5 and 5e-5 is trainable (without gate_multiplier)
When 1e-4 (gate multiplier 10) is collapse 

Need to test 1e-4 without gate_multiplier

**Comment by pureexe** [comment]
Update: 
It failed to to fit single image at 1e-4 both with depth and  depth + normal

![image](https://github.com/user-attachments/assets/5c6b836c-79f4-416c-b30b-5751924cc6b9)

**Comment by pureexe** [comment]
Update: Even without controlnet on training code, the problem still occur. So, the problem might be on how I apply the controller on predicting image
![image](https://github.com/user-attachments/assets/159a8f76-a16e-4017-bf0b-ef8b84058994)

**Comment by pureexe** [comment]
Without controlnet, The code at 2024-08-24 cant be fit with single image. 
However,  the new code supported ControlNet, even when disabling ControlNet. The image looks collapse. So, this is obviously code bug

![image](https://github.com/user-attachments/assets/de202538-3ff5-4ef5-a8f5-c357772017ca)

**Comment by pureexe** [comment]
With old validation code and new training code. The output still collapses.  This suggest that problem is lied in the training loop. I will try to run old training loop with new validation that support controlnet instead

![image](https://github.com/user-attachments/assets/42cce780-8a3e-42cc-b600-b83459ec5724)

**Comment by pureexe** [comment]
With old function compute_train_loss, the problem still occur. 

Still not sure which part of code make this bug

![image](https://github.com/user-attachments/assets/f6475662-24bd-443f-83cf-8a4bedc36bdd)

**Comment by pureexe** [comment]
Problem solve! 
I found out that the problem came from the new code setting the scheduler to UniPC, while the previous code didn't explicitly mention the scheduler. This scheduler comes from the code that is copied from the huggingface demo.

**Comment by pureexe** [comment]
To test the Depth/Normal is improve over the standard version 

We are going to test on 2 method 
1. Testing on Face to get the validation score in term of percent 
2. Test on the shoe to get visualization which we are going to test more method 
2.1 we are going to test the shoe with the point light source. but this might not be the best since it never sees a point of light in training
2.2 we test shoe with the environment map in training set to see if it able to be relight

**Comment by pureexe** [comment]
Now our code is support depth/normal controlnet

**Comment by pureexe** [comment]
We might try to change depth approximation from midas to the better depth approximation
